src/main/java/com/atharvadholakia/password_manager/repository/CredentialRepositoryImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.atharvadholakia.password_manager.repository;
+
+import com.atharvadholakia.password_manager.data.Credential;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class CredentialRepositoryImpl implements CredentialRepository {
+
+  public static final String DATA_FILE =
+      "C:\\Atharva\\Java\\Password_manager\\pwm-backend\\src\\main\\java\\com\\atharvadholakia\\password_manager\\r"
```

Comment:
This static path definition is hard-coded and machine-specific, making it unsuitable for use across different environments. Please make the path dynamic.
1) Move the credentials.json to `resources/`
2) Use dynamic paths. Check online for possible options.